.circleci/config.yml

```diff
@@ -41,22 +41,39 @@ commands:
             release_type=$(echo "$release_tag" | sed 's|\(.*\)_v\([0-9\.]*\).*|\1|')
             echo "export release_type=$release_type" >> $BASH_ENV
 
-jobs:
-  build:
-    working_directory: ~/code
-    docker:
-      - image: circleci/android:api-27-alpha
-    environment:
-      JVM_OPTS: -Xmx3200m
+            newsdkversion=${NEW_SDK_VERSION}
+            echo "export newsdkversion=$newsdkversion" >> $BASH_ENV
+
+
+  setup_emulator:
+    description: >-
+      setup emulator
     steps:
-      - checkout
       - run:
-          name: build the whole project
+          name: Setup emulator
           command: |
-            echo "id=${android_distribution_id}"
-            bash gradlew build -x test
+            sdkmanager "system-images;android-24;default;armeabi-v7a" && echo "no" | avdmanager create avd -n test -k "system-images;android-24;default;armeabi-v7a"
+      - run:
+          name: Launch emulator
+          command: export LD_LIBRARY_PATH=${ANDROID_HOME}/emulator/lib64:${ANDROID_HOME}/emulator/lib64/qt/lib && emulator64-arm -avd test -no-boot-anim -no-window -accel auto -verbose
+          background: true
+      - run:
+          name: Launch logcat
+          command: 
+            adb logcat > logcat.log
+          background: true
+      - run:
+          name: Wait emulator
+          command: |
+            echo "wait for emulator to have booted"
+            circle-android wait-for-boot
+            echo "unlock the emulator screen"
+            sleep 30
```

Comment:
Can this be a loop that checks if the emulator is ready? Say, every 5 seconds.

Reply:
done !